### PR TITLE
Feat/freeze account

### PR DESCRIPTION
**Pull request by mariusmihaic**

### CODE REVIEW

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — DISMISSED** [review]

File: mock/enableEpochsHandlerStub.go [hunk 1]
```diff
@@ -48,6 +50,14 @@ type EnableEpochsHandlerStub struct {
 	StorageAPICostOptimizationEnableEpochField           uint32
 }
 
+func (stub *EnableEpochsHandlerStub) IsGuardAccountEnabled() bool {
```
Comment (sasurobert) [inline mock/enableEpochsHandlerStub.go:53]: comments on public functions

File: builtInFunctions/baseAccountGuarder.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	if check.IfNil(receiverAccount) {
+		return fmt.Errorf("%w for receiver", ErrNilUserAccount)
+	}
+	if vmInput == nil {
+		return ErrNilVmInput
+	}
+
+	senderIsNotCaller := !bytes.Equal(senderAccount.AddressBytes(), vmInput.CallerAddr)
+	senderIsNotReceiver := !bytes.Equal(senderAccount.AddressBytes(), receiverAccount.AddressBytes())
+	if senderIsNotCaller || senderIsNotReceiver {
```
Comment (sasurobert) [inline builtInFunctions/baseAccountGuarder.go:73]: there is no need for such check.

File: builtInFunctions/baseAccountGuarder.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+func (baf *baseAccountGuarder) checkBaseAccountGuarderArgs(
+	senderAccount,
+	receiverAccount vmcommon.UserAccountHandler,
+	vmInput *vmcommon.ContractCallInput,
+	expectedNoOfArgs uint32,
+) error {
+	if check.IfNil(senderAccount) {
+		return fmt.Errorf("%w for sender", ErrNilUserAccount)
+	}
+	if check.IfNil(receiverAccount) {
```
Comment (sasurobert) [inline builtInFunctions/baseAccountGuarder.go:64]: you should not care about the receiverAccount at all. the only check you need is vmInput.CallerAddr == vmInput.RecvAddress

File: builtInFunctions/setGuardian.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+		baseAccountGuarder: base,
+	}
+	setGuardianFunc.activeHandler = args.EnableEpochsHandler.IsSetGuardianEnabled
+
+	return setGuardianFunc, nil
+}
+
+// ProcessBuiltinFunction will process the set guardian built-in function call
+func (sg *setGuardian) ProcessBuiltinFunction(
+	acntSnd, acntDst vmcommon.UserAccountHandler,
```
Comment (sasurobert) [inline builtInFunctions/setGuardian.go:41]: in a set of built in funcitons, where we verifiy that sender == receiver we do not use the acntDst at all. Check ESDTNFTCreate for example

File: builtInFunctions/baseAccountGuarder.go (hunk 2 above)
Comment (iulianpascalau) [reply builtInFunctions/baseAccountGuarder.go:73]: refactored & simplified the code here https://github.com/multiversx/mx-chain-vm-common-go/pull/193

File: builtInFunctions/baseAccountGuarder.go (hunk 3 above)
Comment (iulianpascalau) [reply builtInFunctions/baseAccountGuarder.go:64]: refactored & simplified the code here https://github.com/multiversx/mx-chain-vm-common-go/pull/193

File: builtInFunctions/setGuardian.go (hunk 4 above)
Comment (iulianpascalau) [reply builtInFunctions/setGuardian.go:41]: refactored & simplified the code here https://github.com/multiversx/mx-chain-vm-common-go/pull/193

**Review by sasurobert — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**